### PR TITLE
types: update variadic types

### PR DESCRIPTION
**Pull request by ChristianMurphy**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -53,10 +53,11 @@
 
 /**
  * Create new middleware.
+ * @template MainOutputValue, RestOutputValue
```
Comment (ChristianMurphy) [inline index.js:56]: main and rest output values are shared within this scope

File: index.js [hunk 2]
```diff
@@ -67,8 +68,8 @@ export function trough() {
   /**
    * Call all middleware.
    *
-   * @template MainOutputValue, RestOutputValue
-   * @param {[Callback<MainOutputValue, RestOutputValue>] | [...unknown[], Callback<MainOutputValue, RestOutputValue>]} values
+   * @template Value
+   * @param {[...Value[], Callback<MainOutputValue, RestOutputValue>]} values
```
Comment (ChristianMurphy) [inline index.js:72]: Not :100: if `Value` generic is needed, but it doesn't seem to hurt :man_shrugging:

File: index.js [hunk 3]
```diff
@@ -87,7 +88,7 @@ export function trough() {
      *
      * @template MainOutputValue, RestOutputValue
      * @param {Error?} error
-     * @param {[MainOutputValue] | [MainOutputValue, ...RestOutputValue[]]} [output]
+     * @param {[MainOutputValue?, ...RestOutputValue[]]} [output]
```
Comment (ChristianMurphy) [inline index.js:91]: rest types of 0-n in length.
From usage elsewhere it appears `MainOutputValue` is optional?

**Review by wooorm — APPROVED** [review]
nice, that looks better!

**Pull request merged**